### PR TITLE
[RTL8720E] Add loadable_apps configuration for rtl8720e

### PR DESCRIPTION
**Pull request by thapav**
This patch enables the loadable apps configuration for amebalite

Signed-off-by: Vidisha Thapa <thapa.v@samsung.com>

### CODE REVIEW

**Review by sunghan-chang — COMMENTED** [review]
It looks it includes several kinds of changes. Could you split commits?

File: build/configs/rtl8720e/scripts/rlx8720e_img2_is.ld [hunk 1]
```diff
@@ -229,9 +234,9 @@ SECTIONS
 		. = ALIGN (32);
 	} > KM4_BD_PSRAM_NS_NTZ
 
-	.ARM.extab  :
+	.ARM.extab  : 
 	{
-		*(.ARM.extab* .gnu.linkonce.armextab.*)
+		*(.ARM.extab* .gnu.linkonce.armextab.*)
```
Comment (hdotkashyap) [inline build/configs/rtl8720e/scripts/rlx8720e_img2_is.ld:239]: Please remove whitespaces

File: os/ProtectedLibs.mk [hunk 2]
```diff
@@ -153,7 +153,7 @@ TINYARALIBS += $(LIBRARIES_DIR)$(DELIM)lib_wps$(LIBEXT)
 endif
 
 ifeq ($(CONFIG_AMEBALITE_WIFI),y)
-TINYARALIBS += $(LIBRARIES_DIR)$(DELIM)lib_enc$(LIBEXT)
+#TINYARALIBS += $(LIBRARIES_DIR)$(DELIM)lib_enc$(LIBEXT)
```
Comment (kishore-sn) [inline os/ProtectedLibs.mk:156]: Pls add a space after #

File: os/arch/arm/src/amebalite/Make.defs [hunk 3]
```diff
@@ -178,6 +178,9 @@ CHIP_CSRCS += amebalite_lowerhalf.c
 CHIP_CSRCS += amebalite_timer_lowerhalf.c
 CHIP_CSRCS += amebalite_serial.c
 CHIP_CSRCS += ameba_enet.c
+ifeq ($(CONFIG_BOARDCTL_RESET),y)
+CHIP_CSRCS += amebalite_systemreset.c
+endif
```
Comment (kishore-sn) [inline os/arch/arm/src/amebalite/Make.defs:183]: These changes are not related to loadable apps. Can you pls split into different commits?

File: build/configs/rtl8720e/scripts/rlx8720e_img2_is.ld [hunk 4]
```diff
@@ -241,10 +246,11 @@ SECTIONS
 		*(.ARM.exidx* .gnu.linkonce.armexidx.*)
 		__exidx_end = .;
 	} > KM4_BD_PSRAM_NS_NTZ
-
 	/*-----------------*/
+
 	.ram_image2.bss (NOLOAD):
 	{
+		_sbss = ABSOLUTE(.);
```
Comment (kishore-sn) [inline build/configs/rtl8720e/scripts/rlx8720e_img2_is.ld:253]: Do we not require _ebss?
How about adding _sxxxxx and _exxxx for each section? That is the standard way used in TizenRT.

Comment (thapav) [reply build/configs/rtl8720e/scripts/rlx8720e_img2_is.ld:253]: We have _ebss already. Hence only added _sbss. 
I will verify for all other sections once again. Thank you

File: os/ProtectedLibs.mk (hunk 2 above)
Comment (thapav) [reply os/ProtectedLibs.mk:156]: Updated.

**Review by sunghan-chang — APPROVED** [review]

File: os/ProtectedLibs.mk [hunk 5]
```diff
@@ -153,7 +153,7 @@ TINYARALIBS += $(LIBRARIES_DIR)$(DELIM)lib_wps$(LIBEXT)
 endif
 
 ifeq ($(CONFIG_AMEBALITE_WIFI),y)
-TINYARALIBS += $(LIBRARIES_DIR)$(DELIM)lib_enc$(LIBEXT)
+# TINYARALIBS += $(LIBRARIES_DIR)$(DELIM)lib_enc$(LIBEXT)
```
Comment (sunghan-chang) [inline os/ProtectedLibs.mk:156]: @gexuyan @fengyunlai @ziyik Could you review and confirm this change?

**Comment by sunghan-chang** [comment]
@gexuyan @fengyunlai @ziyik Once you review and confirm my comments above, I will merge this for loadable build. After this, we can go next step.

File: os/ProtectedLibs.mk (hunk 5 above)
Comment (gexuyan) [reply os/ProtectedLibs.mk:156]: Dear @sunghan-chang , This change is ok as it just follows the FlatLibs.mk file, and I have confirmed with Thapa. Thank you.

File: os/board/rtl8720e/src/component/wifi/wifi_interactive_mode.c [hunk 6]
```diff
@@ -811,10 +811,12 @@ int8_t cmd_wifi_on(WiFi_InterFace_ID_t interface_id)
 	wifi_show_setting(WLAN0_NAME, &setting);
 
 #if CONFIG_LWIP_LAYER
-		uint8_t *mac = LwIP_GetMAC(0);
+#ifdef CONFIG_NET_ETHERNET // TO BE UPDATED
```
[the remark this answers is not in the document]
Comment (gexuyan) [reply os/board/rtl8720e/src/component/wifi/wifi_interactive_mode.c:814]: @sunghan-chang , Yes, PR #5578 will resolve this. Revert this change will lead to loadableapps config build fail, then merge #5578 should be able to solve the build error. Or I can rebase my PR #5578 after this PR being merged. Thank You.

File: os/board/rtl8720e/src/component/soc/amebalite/fwlib/ram_km4/ameba_app_start.c [hunk 7]
```diff
@@ -40,24 +40,14 @@ static u32 g_mpu_nregion_allocated;
 extern int main(void);
 extern u32 GlobalDebugEnable;
 
+#ifdef CONFIG_PLATFORM_TIZENRT_OS
```
[the remark this answers is not in the document]
Comment (ziyik) [reply os/board/rtl8720e/src/component/soc/amebalite/fwlib/ram_km4/ameba_app_start.c:43]: Yes, it is okay.
Line 49 is redundant code, it can be removed.

File: os/board/rtl8720e/src/component/wifi/wifi_interactive_mode.c (hunk 6 above)
[the remark this answers is not in the document]
Comment (sunghan-chang) [reply os/board/rtl8720e/src/component/wifi/wifi_interactive_mode.c:814]: @gexuyan To keep build success with this PR, let's merge as it is, and then please rebase yours. Thank you.

**Review by sunghan-chang — APPROVED** [review]

**Comment by sunghan-chang** [comment]
@thapav Please check the conflict.

File: os/board/rtl8720e/src/component/wifi/wifi_interactive_mode.c (hunk 6 above)
[the remark this answers is not in the document]
Comment (gexuyan) [reply os/board/rtl8720e/src/component/wifi/wifi_interactive_mode.c:814]: @sunghan-chang sure no problem
